promql/promqltest/test.go

```diff
@@ -376,6 +413,17 @@ func (t *test) parseEval(lines []string, i int) (int, *evalCmd, error) {
 			break
 		}
 
+		// This would still allow a metric named 'expect' if it is written as 'expect{}'.
+		if strings.Split(defLine, " ")[0] == "expect" {
+			annoType, expectedAnno, err := parseExpect(defLine)
+			if err != nil {
+				return i, nil, formatErr("%w", err)
+			}
+			cmd.expectedCmds[annoType] = append(cmd.expectedCmds[annoType], expectedAnno)
+			j--
+			continue
```

Comment:
Should we check that we don't have conflicting expectation types? For example, a test with both `expect info` and `expect no_info` makes no sense and is likely a mistake.

Reply:
Since this would lead to a failed case, the user will notice it. @beorn7 , what's your take on this?

Reply:
Yeah, I guess the error messages will be pretty clear. I think we should leave things here as is for now. Conflicting expectations could happen on other levels as well, like expecting two different fail messages. Where would we stop?

Reply:
So should we reject multiple failure expectations and all other conflicts?

Reply:
I would indeed try (what I said above: reject multiple failure expectations, reject no_warn together with warn, reject no_info together with info).

Reply:
I have added these checks. Please have a look again.

Reply:
LGTM, thanks.